src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,121 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int CAPACITY = 10;
+    private static final double RESIZE_FACTOR = 1.5;
+    private static final int ELEMENT_NOT_FOUND = -1;
+    private int size;
+    private Object[] array = new Object[CAPACITY];
+
     @Override
     public void add(T value) {
-
+        growIfArrayFull();
+        array[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index < 0 || index > size) {
+            throw new ArrayListIndexOutOfBoundsException("Index " + index
+                    + " out of bounds for size " + size);
+        }
+        growIfArrayFull();
 
+        System.arraycopy(array, index, array, index + 1, size - index);
+        array[index] = value;
+        size++;
     }
```

Comment:
The code for checking the index and throwing an exception if it's out of bounds is repeated in several methods (`add(T value, int index)`, `get(int index)`, `set(T value, int index)`, `remove(int index)`). You should move this logic to a separate private method to avoid code duplication.